skyportal/handlers/api/source.py

```diff
@@ -752,8 +758,23 @@ def get(self, obj_id):
             queries_issued=0,
         )
 
+        self.push_notification(
+            'Finding chart generation in progress. Download will start soon.'
+        )
+        rez = await IOLoop.current().run_in_executor(None, finder)
+
         filename = rez["name"]
-        image = rez["data"]
+        image = io.BytesIO(rez["data"])
+
+        # Adapted from
+        # https://bhch.github.io/posts/2017/12/serving-large-files-with-tornado-safely-without-blocking/
+        mb = 1024 * 1024 * 1
+        chunk_size = 1 * mb
+        max_file_size = 5 * mb
```

Comment:
These can be pretty large if someone chooses a large image size window in the finder. I'd make this max size = 15 Mb

Reply:
Sure, np

Reply:
Done